global_planner/src/orientation_filter.cpp

```diff
@@ -97,6 +97,28 @@ void OrientationFilter::processPath(const geometry_msgs::PoseStamped& start,
                 }
             }
             break;
+        case LEFTRIGHTWARD:
+            for (int i = 0; i < n - 1; i++) {
+                setAngleBasedOnPositionDerivative(path, i);
+            }
+            if (n > 2) {
+                const int num_skips = n >= 5 ? 2 : 1;
+                const double start_path_theta = tf2::getYaw(path[num_skips].pose.orientation);
+                const double goal_path_theta = tf2::getYaw(path[n - 1 - num_skips].pose.orientation);
+                const double start_orientation = tf2::getYaw(start.pose.orientation);
+                const double rotation_to_path = angles::shortest_angular_distance(start_orientation, start_path_theta);
+                const double rotation_to_goal = angles::shortest_angular_distance(start_orientation, goal_path_theta);
+
+                bool prefer_leftward = std::fabs(rotation_to_path - M_PI_2) + std::fabs(rotation_to_goal - M_PI_2) < 
+                                       std::fabs(rotation_to_path + M_PI_2) + std::fabs(rotation_to_goal + M_PI_2);
```

Comment:
F: I think the second time, the sign of the half rotation needs to be inverted, no? I.e. 

```suggestion
                bool prefer_leftward = std::fabs(rotation_to_path - M_PI_2) + std::fabs(rotation_to_goal + M_PI_2) < 
                                       std::fabs(rotation_to_path + M_PI_2) + std::fabs(rotation_to_goal - M_PI_2);
```